### PR TITLE
Disallow empty updates and add test for empty selection list

### PR DESCRIPTION
**Pull request by suresh-prakash**
## Description
The new update method validates if there is at least one update selected to avoid misusing the method without udpates.

### Testing
Added unit test and integration test for empty selections

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java [hunk 1]
```diff
@@ -472,6 +469,11 @@ public Optional<Document> update(
       final org.hypertrace.core.documentstore.query.Query query,
       final java.util.Collection<SubDocumentUpdate> updates)
       throws IOException {
+
+    if (updates.isEmpty()) {
```
Comment (skjindal93) [inline document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java:473]: I hope it doesn't break any existing updates from upstream services

**Review by skjindal93 — APPROVED** [review]

File: document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java (hunk 1 above)
Comment (suresh-prakash) [reply document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java:473]: That's correct. No consumers as of now.

File: document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java [hunk 2]
```diff
@@ -1167,7 +1172,7 @@ private String getColumnValue(
       }
 
       // check for ID column
-      if (Set.of(ID, IMPLICIT_ID_ALIAS).contains(columnName)) {
+      if (OUTER_COLUMNS.contains(columnName) || IMPLICIT_ID_ALIAS.equals(columnName)) {
```
Comment (kotharironak) [inline document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java:1175]: What was the issue? Wasn't the existing condition sufficient - `Set.of(ID, IMPLICIT_ID_ALIAS).contains(columnName)`?

**Review by kotharironak — COMMENTED** [review]
LGTM

File: document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java [hunk 3]
```diff
@@ -1167,7 +1172,7 @@ private String getColumnValue(
       }
 
       // check for ID column
```
Comment (kotharironak) [inline document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java:1174]:
```suggestion
      // check for any OUTER column including ID, and its alias
```

Comment (kotharironak) [reply document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java:1174]: If, we need this - https://github.com/hypertrace/document-store/pull/132#discussion_r984253529, let's change the comment

File: document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java (hunk 2 above)
Comment (kotharironak) [reply document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java:1175]: Discussed offline, it was needed with an empty selection.

File: document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java (hunk 3 above)
Comment (suresh-prakash) [reply document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java]: Done.

**Review by kotharironak — DISMISSED** [review]

File: document-store/src/integrationTest/java/org/hypertrace/core/documentstore/DocStoreQueryV1Test.java [hunk 4]
```diff
@@ -1366,6 +1366,27 @@ public void testFindWithKeyAndNonMatchingRelationalFilter(final String datastore
...
+  @ParameterizedTest
+  @MethodSource("databaseContextBoth")
+  public void testFindWithMultipleKeysAndPartiallyMatchingRelationalFilter(
+      final String datastoreName) throws IOException {
+    final Collection collection = getCollection(datastoreName);
+    final org.hypertrace.core.documentstore.query.Filter filter =
+        org.hypertrace.core.documentstore.query.Filter.builder()
+            .expression(
+                and(
+                    or(
```
Comment (kotharironak) [inline document-store/src/integrationTest/java/org/hypertrace/core/documentstore/DocStoreQueryV1Test.java:1378]: This is a good example. Thanks for adding this one.

**Review by kotharironak — APPROVED** [review]
